src/Symfony/Component/Messenger/Message/LockableMessage.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Message;
+
+use Symfony\Component\Lock\Key;
+
+interface LockableMessage
+{
+    /**
+     * Returns null if you want to force the dispatch of the message.
+     */
+    public function getKey(): ?Key;
```

Comment:
I think this shold be `function getLockKey(): ?string;`

The complexity (and coupling with symfony/lock) of generating the Key, chould be handled by the Middleware.
That would provide a simpler/easier to implement interface for end-users.